WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.Extensions.Hosting;
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class AliceClient : BackgroundService, IAsyncDisposable
+	{
+		private Guid AliceId { get; }
+		private Guid RoundId { get; }
+		private ArenaClient ArenaClient { get; }
+		public IEnumerable<ICoin> CoinsToRegister { get; }
+
+		public AliceClient(Guid aliceId, Guid roundId, ArenaClient arenaClient, IEnumerable<ICoin> coinsToRegister)
+		{
+			AliceId = aliceId;
+			RoundId = roundId;
+			ArenaClient = arenaClient;
+			CoinsToRegister = coinsToRegister;
+		}
+
+		protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+		{
+			// Input registration phase
+			do
+			{
+				await Task.Delay(TimeSpan.FromSeconds(30), stoppingToken).ConfigureAwait(false);
+				await ConfirmationAsync().ConfigureAwait(false);
```

Comment:
i guess this is just because it's a sketch but the loop should be exited successfully when a successful response is received, that's still in scope right?

Reply:
Yes, I think so. The problem is `ArenaClient::ConfirmConnection` method is not returning anything when it really should return and Phase or Bool to leat us know we are in Connection Confirmation phase and we can exit the loop. A boolean is okay I think.

Reply:
ACK